### PR TITLE
Rebuild with Next.js

### PR DESCRIPTION
**Pull request by TheoBr**
Ready to go :)

Deployed at [osrg.t3.gg](https://osrg.t3.gg)

### Things I need to do still
- [x] Link to Discord
- [x] Add ci/build checks
- [x] Add tRPC for api definitions
- [x] Rebuild contact form

### Things I want to do still
- [x] Fancier homepage
- [x] Custom assets for favicon, social embeds, etc

### Things I need help on
- [x] Wtf do the other pages do?
- [x] Copy for home page + about page
- [x] Plan of attack for "contact form", not sure how you want to do that. Can I just make an endpoint that pings the discord server?

### CODE REVIEW

**Comment by JacobMGEvans** [comment]
You're a beast.

**Comment by nobrayner** [comment]
>  Wtf do the other pages do?

Not much 😅  - the only one that's actually useful is the raids pages (in which it displays previous/current raids, and then an individual raid page shows the stats for that raid)

>  Plan of attack for "contact form", not sure how you want to do that. Can I just make an endpoint that pings the discord server?

@JacobMGEvans discord ping is probably good enough, yeah? Going to potential targets?

**Comment by JacobMGEvans** [comment]
> > Wtf do the other pages do?
> 
> Not much 😅 - the only one that's actually useful is the raids pages (in which it displays previous/current raids, and then an individual raid page shows the stats for that raid)
> 
> > Plan of attack for "contact form", not sure how you want to do that. Can I just make an endpoint that pings the discord server?
> 
> @JacobMGEvans discord ping is probably good enough, yeah? Going to potential targets?

Yeah currently it sends a Form object to Netlify, but that was just the best solution at the time.

**Comment by TheoBr** [comment]
@JacobMGEvans As an ex-user of Netlify's built-in forms, I can confirm they were _never the best option for anything_

Discord bot is added and should be good to go :) If we find it gets noisy I think we should add Github auth in front of it

File: src/backend/utils/ping-discord.ts
Comment (nobrayner) [inline src/backend/utils/ping-discord.ts:3]: Shouldn't this be a secret or something? 😅

Comment (TheoBr) [reply src/backend/utils/ping-discord.ts:3]: I thought I left a comment above saying such lol my bad, can generate a new one and kill this link

Comment (TheoBr) [reply src/backend/utils/ping-discord.ts]: Updated accordingly :)

**Review by nobrayner — DISMISSED** [review]

**Comment by TheoBr** [comment]
We should probably make another repo for the Netlify deploy that redirects to the new domain before merging

File: .eslintrc.json
Comment (JacobMGEvans) [inline .eslintrc.json:1]: Some of this I will probably put back.

File: .vscode/settings.json
Comment (JacobMGEvans) [inline .vscode/settings.json:2]: I generally run the formatter through ESLint

File: README.md
Comment (JacobMGEvans) [inline README.md:10]: Nice I was planning on switching to this too!

File: prisma/schema.prisma
Comment (JacobMGEvans) [inline prisma/schema.prisma:1]: wow the schema and migration is simpler than I expected 🤯

File: .eslintrc.json
Comment (TheoBr) [reply .eslintrc.json:1]: I'm indifferent tbh. Only reason I "changed" it is because I nuked the original dir and used the `next lint` defaults 😆 

Generally I like to "use the defaults of someone who cares more" for things like this, Prettier, TSConfig, etc, but if you have stronger preferences go nuts

File: .vscode/settings.json
Comment (TheoBr) [reply .vscode/settings.json:2]: ESLint w/ prettier plugin I'm assuming? Makes sense, happy to switch that over (but would prefer you do it since I don't know how it was configured before and I _genuinely don't give a shit about linting lmao_)

File: src/backend/router/index.ts
Comment (JacobMGEvans) [inline src/backend/router/index.ts:5]: "server side" in quotes  lmao

File: prisma/schema.prisma
Comment (TheoBr) [reply prisma/schema.prisma:1]: Right? It's a _tad_ complex for my liking, but with full user-OAuth account bindings, multi-account sign in and session management it's pretty tiny

**Review by JacobMGEvans — APPROVED** [review]

File: prisma/schema.prisma
Comment (balazsorban44) [reply prisma/schema.prisma:1]: with a custom adapter you could drop some of these fields as well. we just forward all fields for simplicity on our side. account has multiple potentially unused fields, but you could drop them in your own adapter. the default Prisma adapter is 48LOC, so wouldn't be that much work either. https://github.com/nextauthjs/adapters/blob/next/packages/prisma/src/index.ts

**Pull request merged**
